### ISSUE TITLE
[Clang][RISC-V] SiFive CLIC preemptible interrupt handlers fail with a backend fatal error at -O0

### ISSUE THREAD

**Issue by zeyi2** [llvm/llvm-project#217936]
Clang enables frame pointers by default at `-O0`, while the current SiFive CLIC implementation does not support frame pointers for preemptible interrupt handlers. Instead of producing a normal diagnostic, Clang prints a crash stack trace and asks the user to report an LLVM bug.

This reproduces for both RV32 and RV64.

Reproducer: https://godbolt.org/z/c6W4GYzh5

The [SiFive Interrupt Cookbook v1.2](https://sifive.cdn.prismic.io/sifive/d1984d2b-c9b9-4c91-8de0-d68a5e64fa0f_sifive-interrupt-cookbook-v1p2.pdf), section 3.2.4, explicitly states:

```
Using the SiFive-CLIC-preemptible attribute requires the addition of the -fomit-frame-pointer compiler flag.
```

Therefore, retaining the current frame-pointer restriction seems reasonable, but it should be reported through a normal diagnostic rather than `reportFatalUsageError`:

I think at minimum, it should report a normal compilation error such as:

```
error: 'SiFive-CLIC-preemptible' interrupt handlers do not support frame pointers; compile with '-fomit-frame-pointer'
```

**Comment by zeyi2** [comment]
CC @lenary, WDYT? Does the proposed approach seem reasonable? I'd be happy to open a PR to address the diagnostic.

**Comment by topperc** [comment]
-fomit-frame-pointers won't stop all uses of a frame pointer. So we might need to diagnose why a frame pointer is needed.

Placing a variable sized object on the stack requires a frame pointer https://godbolt.org/z/dbaqzK3ez

So does aligning a variable to more than 16 bytes. https://godbolt.org/z/coEhz4MW4

**Comment by lenary** [comment]
Yeah this is going to be "fun" to fix. 

I think an error at the moment is good, though craig is right to mention that we cannot point to a user option causing a frame pointer.

We're going to have to do a lot better when we do codegen support for the new interrupt support for Smip/Ssip.
